php/utils.php

```diff
@@ -884,3 +886,28 @@ function expand_globs( $paths, $flags = GLOB_BRACE ) {
 
 	return array_unique( $expanded );
 }
+
+/**
+ * Get a Phar-safe version of a path.
+ *
+ * For paths inside a Phar, this strips the outer filesystem's location to
+ * reduce the path to what it needs to be within the Phar archive.
+ *
+ * Use the __FILE__ or __DIR__ constants as a starting point.
+ *
+ * @param string $path An absolute path that might be within a Phar.
+ *
+ * @return string A Phar-safe version of the path.
+ */
+function phar_safe_path( $path ) {
+
+	if ( ! inside_phar() ) {
+		return $path;
+	}
+
+	return str_replace(
+		PHAR_STREAM_PREFIX . getcwd() . '/',
```

Comment:
> The actual paths you want to use depend on the package you're in, and how the packages have been included. So, I'm not sure we can reduce this problem to 1 single constant.

The packages care about the _base path_ for the Phar file. Setting a `WP_CLI_PHAR_PATH` at the beginning of the Phar execution will provide this value, more consistently than `getcwd()` can.

> If this works as expected in all scenarios, then it works across the board for getting rid of all the "three-different-ways-of-getting-a-path" issues.

What if my script calls `chdir()`?